### PR TITLE
nxos_logging implementation module

### PR DESCRIPTION
**Pull request by trishnaguha**
Signed-off-by: Trishna Guha <trishnaguha17@gmail.com>

##### SUMMARY

nxos_logging
net_logging implementation

##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

modules/network/nxos

##### ANSIBLE VERSION

```
devel 2.4
```

### CODE REVIEW

File: lib/ansible/modules/network/nxos/nxos_logging.py [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+
+    return obj
+
+
+def main():
+    """ main entry point for module execution
+    """
+    argument_spec = dict(
+        dest=dict(type='str', choices=DEST_GROUP),
+        name=dict(type='str'),
```
Comment (ganeshrn) [inline lib/ansible/modules/network/nxos/nxos_logging.py]: type='str' is a default value not required to mention in separately.

**Review by ganeshrn — APPROVED** [review]
Apart from a minor change in `argument_spec ` looks good to me.

**Comment by os4mike** [comment]
How do you use the nxos_logging 'aggregate' command. There are no examples to find, or am I missing something?

https://docs.ansible.com/ansible/2.4/nxos_logging_module.html

**Comment by trishnaguha** [comment]
@os4mike 
```
  nxos_logging:
    aggregate:
      - { dest: logfile, name: test, dest_level: 0 }
      - { facility: daemon, facility_level: 0 }
    state: absent
    provider: "{{ connection }}"
```

**Comment by os4mike** [comment]
Thanks Trishna, The Nexus logging configuration is fairly complicated - we require addition of the following line of config-

    logging server 192.168.1.1 5 use-vrf default

Is this a possibility at this stage in Ansible, or would the nxos_config module be a better choice to implement this line?

**Comment by trishnaguha** [comment]
@os4mike since nxos logging configuration is fairly complicated and has plenty of options to be considered which is not always possible, you might want to use nxos_config in order to achieve what you want for now. 

It also looks like syslog server doesn't exist as destination option for nxos_logging module. But it sounds like it should be there. Would you mind opening an issue for that? We will try to get it done by 2.5. Thanks!

**Pull request merged**